Add package QWT 5.2.1 to Buildroot. An option enables the build of the examples.

Signed-off-by: Jérémie Scheer <jeremie.scheer@armadeus.com>

Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2012-02-06 17:00:50.000000000 +0100
+++ buildroot/package/Config.in	2012-02-06 17:05:03.000000000 +0100
@@ -134,6 +134,7 @@
 
 comment "other GUIs"
 source "package/qt/Config.in"
+source "package/qwt/Config.in"
 source "package/x11r7/Config.in"
 
 comment "X libraries and helper libraries"
Index: buildroot/package/qwt/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/qwt/Config.in	2012-02-06 17:00:50.000000000 +0100
@@ -0,0 +1,20 @@
+comment "Qwt requires Qt to be installed"
+	depends on !BR2_PACKAGE_QT
+
+menuconfig BR2_PACKAGE_QWT
+	bool "Qwt"
+	depends on BR2_PACKAGE_QT
+	help
+	  Qwt for Embedded Linux.
+
+	  http://qwt.sourceforge.net/
+
+if BR2_PACKAGE_QWT
+
+config BR2_PACKAGE_QWT_EXAMPLES
+	bool "Examples"
+	select BR2_PACKAGE_QT_SVG
+	help
+	  Compile & install the examples.
+
+endif #BR2_PACKAGE_QWT
Index: buildroot/package/qwt/qwt-5.2.1-change_install_dir_and_no_designer.patch
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/qwt/qwt-5.2.1-change_install_dir_and_no_designer.patch	2012-02-06 17:00:50.000000000 +0100
@@ -0,0 +1,30 @@
+--- qwt-5.2.1/qwtconfig.pri.orig	2010-09-28 14:34:01.000000000 +0200
++++ qwt-5.2.1/qwtconfig.pri	2010-09-28 15:37:02.000000000 +0200
+@@ -115,7 +115,7 @@
+ # Otherwise you have to build it from the designer directory.
+ ######################################################################
+ 
+-CONFIG     += QwtDesigner
++#CONFIG     += QwtDesigner
+ 
+ ######################################################################
+ # If you want to auto build the examples, enable the line below
+
+--- qwt-5.2.1/src/src.pro.orig	2010-09-28 14:40:39.000000000 +0200
++++ qwt-5.2.1/src/src.pro	2010-09-28 14:41:29.000000000 +0200
+@@ -218,9 +218,12 @@
+ }
+ 
+ # Install directives
+-
+-headers.files  = $$HEADERS
+-doc.files      = $${QWT_ROOT}/doc/html $${QWT_ROOT}/doc/qwt-5.2.0.qch
++
++target.path = /usr/lib
++headers.files  = $$HEADERS
++headers.path = /usr/include/qwt-5.2.1
++doc.files      = $${QWT_ROOT}/doc/html $${QWT_ROOT}/doc/qwt-5.2.0.qch
++doc.path = /usr/doc/qwt-5.2.1
+ unix {
+     doc.files      += $${QWT_ROOT}/doc/man
+ }
Index: buildroot/package/qwt/qwt.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/qwt/qwt.mk	2012-02-06 17:06:49.000000000 +0100
@@ -0,0 +1,82 @@
+######################################################################
+#
+# QWT
+#
+######################################################################
+
+QWT_VERSION:=5.2.1
+QWT_SOURCE:=qwt-$(QWT_VERSION).zip
+QWT_SITE:=https://sourceforge.net/projects/qwt/files/qwt/$(QWT_VERSION)
+QWT_UNZIP:=unzip
+QWT_TARGET_DIR:=$(BUILD_DIR)/qwt-$(QWT_VERSION)
+QWT_CONFIG_FILE:=qwt.pro
+QWT_EXAMPLES_CONFIG_FILE:=examples.pro
+QT_DIR:=$(HOST_DIR)/usr
+
+$(DL_DIR)/$(QWT_SOURCE):
+	$(call DOWNLOAD,$(QWT_SITE),$(QWT_SOURCE))
+
+qwt-source: $(DL_DIR)/$(QWT_SOURCE)
+
+
+$(QWT_TARGET_DIR)/.unpacked: $(DL_DIR)/$(QWT_SOURCE)
+	$(QWT_UNZIP) -d $(BUILD_DIR) $(DL_DIR)/$(QWT_SOURCE)
+	toolchain/patch-kernel.sh $(QWT_TARGET_DIR) package/qwt/ \
+		qwt-$(QWT_VERSION)-\*.patch \
+		qwt-$(QWT_VERSION)-\*.patch.$(ARCH)
+	touch $@
+
+$(QWT_TARGET_DIR)/.configured: $(QWT_TARGET_DIR)/.unpacked
+	-[ -f $(QWT_TARGET_DIR)/Makefile ] && $(MAKE) -C $(QWT_TARGET_DIR) distclean
+	$(QT_DIR)/bin/qmake $(QWT_TARGET_DIR)/$(QWT_CONFIG_FILE)
+	touch $@
+
+$(QWT_TARGET_DIR)/.compiled: $(QWT_TARGET_DIR)/.configured
+	$(MAKE) -C $(QWT_TARGET_DIR)
+	touch $@
+
+$(STAGING_DIR)/usr/lib/libqwt.so: $(QWT_TARGET_DIR)/.compiled
+	(export INSTALL_ROOT=$(STAGING_DIR); $(MAKE) -C $(QWT_TARGET_DIR) install)
+
+$(TARGET_DIR)/usr/lib/libqwt.so: $(STAGING_DIR)/usr/lib/libqwt.so
+	cp -dpf $(STAGING_DIR)/usr/lib/libqwt.so* $(TARGET_DIR)/usr/lib/
+	-$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(TARGET_DIR)/usr/lib/libqwt.so*
+
+$(QWT_TARGET_DIR)/examples/.configured:
+	-[ -f $(QWT_TARGET_DIR)/examples/Makefile ] && $(MAKE) -C $(QWT_TARGET_DIR)/examples distclean
+	$(QT_DIR)/bin/qmake $(QWT_TARGET_DIR)/examples/$(QWT_EXAMPLES_CONFIG_FILE)
+	touch $@
+
+$(QWT_TARGET_DIR)/examples/.compiled: $(QWT_TARGET_DIR)/examples/.configured
+	$(MAKE) -C $(QWT_TARGET_DIR)/examples
+	touch $@
+
+$(STAGING_DIR)/usr/local/qwt-5.2.1/examples/bin/bode: $(QWT_TARGET_DIR)/examples/.compiled
+	mkdir -p $(STAGING_DIR)/usr/local/qwt-5.2.1/examples/bin
+	cp -f $(QWT_TARGET_DIR)/examples/bin/* $(STAGING_DIR)/usr/local/qwt-5.2.1/examples/bin
+
+$(TARGET_DIR)/usr/local/qwt-5.2.1/examples/bin/bode: $(STAGING_DIR)/usr/local/qwt-5.2.1/examples/bin/bode
+	mkdir -p $(TARGET_DIR)/usr/local/qwt-5.2.1/examples/bin
+	cp -f $(STAGING_DIR)/usr/local/qwt-5.2.1/examples/bin/* $(TARGET_DIR)/usr/local/qwt-5.2.1/examples/bin
+
+ifneq ($(BR2_PACKAGE_QWT_EXAMPLES),y)
+qwt: qt $(TARGET_DIR)/usr/lib/libqwt.so
+else
+qwt: qt $(TARGET_DIR)/usr/lib/libqwt.so $(TARGET_DIR)/usr/local/qwt-5.2.1/examples/bin/bode
+endif
+
+qwt-clean:
+	-$(MAKE) -C $(QWT_TARGET_DIR) clean
+	-rm $(TARGET_DIR)/usr/lib/libqwt.so.*
+
+qwt-dirclean:
+	rm -rf $(QWT_TARGET_DIR)
+
+#############################################################
+#
+# Toplevel Makefile options
+#
+#############################################################
+ifeq ($(BR2_PACKAGE_QWT),y)
+TARGETS+=qwt
+endif
